lib/presentation/contact_manager/contact_manager.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:async';
+import 'package:dartz/dartz.dart';
+import 'package:debounce_throttle/debounce_throttle.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success_converter.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/config/app_config.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/contact/get_contacts_state.dart';
+import 'package:fluffychat/domain/usecase/get_all_contacts_interactor.dart';
+import 'package:fluffychat/domain/usecase/search_contacts_interactor.dart';
+import 'package:fluffychat/presentation/converters/presentation_contact_converter.dart';
+import 'package:fluffychat/presentation/extensions/contact/presentation_contact_extension.dart';
+import 'package:fluffychat/presentation/model/presentation_contact.dart';
+import 'package:fluffychat/widgets/twake_components/twake_smart_refresher.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+class ContactManager {
+  bool _firstSynchronize = false;
+
+  static const _debouncerIntervalInMilliseconds = 300;
+
+  final _getAllContactsInteractor = getIt.get<GetAllContactsInteractor>();
+
+  final searchContactsInteractor = getIt.get<SearchContactsInteractor>();
+
+  final SuccessConverter converter = PresentationContactConverter();
+
+  // FIXME: Consider can use FocusNode instead ?
+  final ValueNotifier<bool> isSearchModeNotifier = ValueNotifier(false);
+
+  final ValueNotifier<Either<Failure, Success>> contactsNotifier =
+      ValueNotifier<Either<Failure, Success>>(
+    const Right(GetContactsInitial()),
+  );
+
+  final ValueNotifier<List<PresentationContact>> tomPresentationContacts =
+      ValueNotifier<List<PresentationContact>>([]);
+
+  final ValueNotifier<List<PresentationContact>> phonebookPresentationContacts =
+      ValueNotifier<List<PresentationContact>>([]);
+
+  final TwakeRefreshController refreshController = TwakeRefreshController();
+
+  final Debouncer _debouncer = Debouncer(
```

Comment:
what is the debouncer? I mean this debouncer use for what?

Reply:
I think keep `_debouncer ` for search contacts from local. WDYT?